### PR TITLE
Integration tests/features

### PR DESCRIPTION
**Pull request by Tadessejemal1**
# Good Reviewers you are welcome 👍🏼 

## Project checks ✔️ 

in this milestone, we have done the following tasks

- Integration specs

Use Capybara to write integration tests for each view in your project.

- User index page:

1. I can see the username of all other users.
2. I can see the profile picture for each user.
3. I can see the number of posts each user has written.
4. When I click on a user, I am redirected to that user's show page.

- user show page:

1. I can see the user's profile picture.
2. I can see the user's username.
3. I can see the number of posts the user has written.
4. I can see the user's bio.
5. I can see the user's first 3 posts.
6. I can see a button that lets me view all of a user's posts.
7. When I click a user's post, it redirects me to that post's show page.
8. When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:

1. I can see the user's profile picture.
2. I can see the user's username.
3. I can see the number of posts the user has written.
4. I can see a post's title.
5. I can see some of the post's body.
6. I can see the first comments on a post.
7. I can see how many comments a post has.
8. I can see how many likes a post has.
9. I can see a section for pagination if there are more posts than fit on the view.
10. When I click on a post, it redirects me to that post's show page.

- Post show page:

1. I can see the post's title.
2. I can see who wrote the post.
3. I can see how many comments it has.
4. I can see how many likes it has.
5. I can see the post body.
6. I can see the username of each commentor.
7. I can see the comment each commentor left.

## Specs checks ✔️ 

image

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -22,7 +22,8 @@
     end
```
Comment (HAMMAS-SALEEM) [inline spec/requests/users_spec.rb:23]: image

- [ ] One of your test is failing kindly fix that. You will see that by dropping your database and by creating it again :blush:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @angeluray, @Tadessejemal1 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requested changes, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/users_spec.rb (hunk 1 above)
Comment (angeluray) [reply spec/requests/users_spec.rb:23]: Hi @HAMMAS-SALEEM , thank you so much for your feedback. I already solved it 👍🏼

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Tadessejemal1 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**